test/unit/utDefaultIOStream.cpp

```diff
@@ -44,10 +44,20 @@ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #include <string>
 
 #if defined(__GNUC__) || defined(__clang__)
-#define MAKE_TEMP_FILE_NAME mktemp
+#define TMP_PATH "/tmp/"
+void MakeTmpFilePath(char* tmplate)
+{
+    auto err = mkstemp(tmplate);
```

Comment:
This doesn't do what you think it does. `mkstemp` doesn't change `tmplate`, instead it returns a file descriptor. Throwing away its return value at the end of the function is wrong.

Reply:
Maybe you should be using [std::tmpfile](http://en.cppreference.com/w/cpp/io/c/tmpfile) instead?

Reply:
I tried to make std::tmpfile work initially, but the point of the test is to get the file's size, which can't be done without a path to the file.
It seemed like mkstemp was filling in the appropriate placeholders into tmplate when I was printing the result. I'll dig deeper tomorrow.

Reply:
Having looked over some more documentation for mkstemp, I can see a better way of implementing that function. I'll make the necessary improvement tomorrow.

Reply:
@turol I'm slightly confused how it ever worked, considering mkstemp opens the file for exclusive use. Maybe fopen is playing some trick.

Reply:
it was leaving `fpath` untouched so you `fopen`:ed a file called `"./rndfp.XXXXXX"`.

Reply:
http://manpages.ubuntu.com/manpages/xenial/man3/mkstemp.3.html
This seems to suggest that the template is modified. This seems to be the intended behaviour on Linux, as far as I can work out.
I was wrong about mkstemp opening the file for exclusive use, the common behaviour is only to ensure that the calling process is what creates the file in the first place. That explains why opening it again afterwards wasn't failing immediately. I've improved this function regardless. It will be in my next pull request on this issue. @turol